examples/grid2grid/main.py

```diff
@@ -0,0 +1,161 @@
+        provides the corresponding joint angles"""
+        if self.plot:
+            fig = plt.figure(figsize=(10, 5))
+            plt.style.use('seaborn-pastel')
+            ax = [fig.add_subplot(121), fig.add_subplot(122, projection='3d')]
+        _joints = {grids[0]: {'pick': [], 'hover': []}, grids[1]: {'pick': [], 'hover': []}}
+
+        for grid_iter in grids:
+            obj_h = abs(config['grids']['object'][grid_iter])
+            ee_h = abs(config['EVA']['end_effector']['length'])
+            hover_h = abs(config['EVA']['hover_height'])
+            guess = config['grids']['guess'][grid_iter]
+            _grid_x, _grid_y, _grid_z, _grid_points = self._create_grid(grid_iter)
+
+            with ChargingBar('Computing ' + config['grids']['names_verbose'][grid_iter] + ' grid', max=len(_grid_points)) as bar:
+                for _counter in range(len(_grid_points)):
+                    x = _grid_points[_counter][0]/1000
+                    y = _grid_points[_counter][1]/1000
+                    z = (_grid_points[_counter][2]+ee_h+obj_h)/1000  # [m]
+                    pos_obj = [x, y, z]  # [m]
+                    pos_obj_hover = [x, y, z + hover_h]  # [m]
+                    extra_angle = config['grids']['angle_pickup'][grid_iter]     # Additional pickup angle
+                    success_pick, _joints_pick = solve_ik(self.eva, guess, extra_angle, pos_obj)
+                    success_hover, _joints_hover = solve_ik(self.eva, guess, extra_angle, pos_obj_hover)
+                    if ('success' not in success_pick) or ('success' not in success_hover):
+                        raise Exception('IK error')  # Failed IK. Position not reachable
+                    _joints[grid_iter]['pick'].append(_joints_pick)
+                    _joints[grid_iter]['hover'].append(_joints_hover)
+                    bar.next()
+                if self.plot:
+                    plot_grids(_grid_x, _grid_y, _grid_z, ax, grid_iter)
+        if self.plot:
+            plt.show()
+        move_eva = input("Please verify the correctness of grid placement before continuing "
+                         "(to do this, set plot_on_off variable to True and verify grids placement from the plots).\n"
+                         "Proceed (this will move the robot)? yes/no\n")
+        if 'yes' not in move_eva:
+            raise Exception('\n\n\nScript aborted by user: \n - set the plot_on_off variable to True\n '
+                            '- run the script again\n - verify grids placement from the plots ')
+        return _joints
+
+
+if __name__ == "__main__":
+    # Load use-case parameters
+    config = load_use_case_config()
+
+    # Connection to robot
+    host = config['EVA']['comm']['host']
+    token = config['EVA']['comm']['token']
+    eva = Eva(host, token)
+
+    # Compute grid points and robot joints
+    plot_on_off = True
+    eva_box = EvaGrids(eva, plot_on_off)

```

Comment:
If you wanted to make this a bit cleaner, you could not define "_plot_on_off=True"

Reply:
done